nntrainer/layers/loss_layer.cpp

```diff
@@ -129,15 +129,31 @@ void LossLayer::calcDerivative() {
   /// @todo loss backwarding is allocating a new memory. loss layer shouldn't!!
   switch (loss_type) {
   case LossType::LOSS_MSE:
-    ret_derivative = y.subtract(y2).multiply(2).divide(y.getDim().getDataLen());
+    y.subtract(y2, ret_derivative);
+    ret_derivative.multiply_i(2);
+    if (ret_derivative.divide_i(y.length()) != ML_ERROR_NONE) {
+      throw std::runtime_error(
+        "[Loss::calcDerivative] Error when calculating loss");
+    }
+    // ret_derivative = ret_derivative.divide(y.length());
     break;
   case LossType::LOSS_ENTROPY_SIGMOID:
-    ret = y.apply(ActivationLayer::sigmoid, ret);
-    ret_derivative = ret.subtract(y2).divide(ret.getDim().getDataLen());
+    y.apply(ActivationLayer::sigmoid, ret);
+    ret.subtract(y2, ret_derivative);
+    if (ret_derivative.divide_i(ret.length()) != ML_ERROR_NONE) {
+      throw std::runtime_error(
+        "[Loss::calcDerivative] Error when calculating loss");
+    }
+    // ret_derivative = ret.subtract(y2).divide(ret.getDim().getDataLen());
     break;
   case LossType::LOSS_ENTROPY_SOFTMAX:
-    ret = y.apply(ActivationLayer::softmax, ret);
-    ret_derivative = ret.subtract(y2).divide(ret.batch());
+    y.apply(ActivationLayer::softmax, ret);
```

Comment:
Please add a comment here that y and ret_derivative can be same here. 
So, this operation cannot apply in-place.

Reply:
https://github.com/nnstreamer/nntrainer/issues/943#issuecomment-773865415

Reply:
Done